sixtyfps_compiler/generator/cpp.rs

```diff
@@ -621,10 +621,51 @@ fn generate_public_component(file: &mut File, component: &llr::PublicComponent)
         argument_types: &[],
     };
 
-    component_struct.members.iter_mut().for_each(|(access, _)| {
-        *access = Access::Private;
+    let old_declarations = file.declarations.len();
+
+    generate_item_tree(
+        &mut component_struct,
+        &component.item_tree,
+        &component,
+        None,
+        component_id.clone(),
+        file,
+    );
+
+    // Hide properties and other fields from the public C++ API
+
+    let destructor_name = format!("~{}", component_struct.name);
+
+    component_struct.members.iter_mut().for_each(|(access, decl)| {
+        let hide = match decl {
+            Declaration::Var(var @ Var { .. })
+                if var.name != "static_vtable" && var.name != "self_weak" =>
+            {
+                true
+            }
+            Declaration::Function(func @ Function { .. })
+                if func.name != "create" && func.name != destructor_name =>
+            {
+                true
+            }
+            _ => false,
+        };
+        if hide {
+            *access = Access::Private;
+        }
     });
 
```

Comment:
I am sure I will end up adding something with public access later and then wonder why that is generated as private!

Can't we do that when we define those instead of changing the Access behind the back of the programmer?

That should also work if you keep the code that adds the friends below I think.

Reply:
I hope I understood your suggestion correct. I've pushed a new version of the patch.